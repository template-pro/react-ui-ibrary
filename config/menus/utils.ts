const utilsMenu = [
  {
    title: 'General',
    children: ['src/with-default-props'],
  },
  {
    title: 'React Hooks',
    children: ['src/use-props-value'],
  },
  // {
  //   title: "Date & Time",
  //   children: [],
  // },
  // {
  //   title: "String",
  //   children: [],
  // },
]

export default utilsMenu
