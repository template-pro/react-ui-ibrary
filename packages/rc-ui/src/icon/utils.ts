import { useEffect } from 'react'
import { insertCss } from 'insert-css'

export const svgBaseProps = {
  // width: '1em',
  // height: '1em',
  'fill': 'currentColor',
  'aria-hidden': 'true',
  'focusable': 'false',
}

export const iconStyles = `
.template-pro-icon {
  display: inline-block;
  width: 24px;
  color: inherit;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.template-pro-icon > * {
  line-height: 1;
}

.template-pro-icon svg {
  display: inline-block;
}

.template-pro-icon::before {
  display: none;
}

.template-pro-icon .template-pro-icon {
  display: block;
}

.template-pro-icon[tabindex] {
  cursor: pointer;
}

.template-pro-icon-spin::before,
.template-pro-icon-spin {
  display: inline-block;
  -webkit-animation: loadingCircle 1s infinite linear;
  animation: loadingCircle 1s infinite linear;
}

@-webkit-keyframes loadingCircle {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes loadingCircle {
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}
`

let cssInjectedFlag = false

export const useInsertStyles = (styleStr = iconStyles) => {
  useEffect(() => {
    if (!cssInjectedFlag) {
      insertCss(styleStr, {
        prepend: true,
      })
      cssInjectedFlag = true
    }
  }, [])
}
