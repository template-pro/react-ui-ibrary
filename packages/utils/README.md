# @template-pro/utils

统一工具库，通用正则，React Hooks 等工具。

## 安装

```bash
npm install @template-pro/utils
```
