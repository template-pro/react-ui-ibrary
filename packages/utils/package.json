{
  "name": "@template-pro/utils",
  "version": "0.0.0",
  "description": "统一工具库，通用正则，react hooks 等工具",
  "main": "./lib/index.js",
  "module": "./es/index.js",
  "types": "./es/index.d.ts",
  "files": [
    "lib",
    "es"
  ],
  "scripts": {
    "build": "gulp",
    "release":"bumpp --tag @template-pro/utils@%s --all --no-verify && pnpm publish"
  },
  "keywords": [
    "utils"
  ],
  "author": "Wuxh<wxh1220@gmail.com>",
  "license": "MIT",
  "publishConfig": {
    "registry": "http://nas.wxhboy.cn:98/"
  }
}
