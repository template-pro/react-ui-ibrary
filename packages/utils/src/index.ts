export { default as usePropsValue } from './use-props-value'
export { default as mergeProps } from './with-default-props'

// RegExp
export * as anyRule from './any-rule'
