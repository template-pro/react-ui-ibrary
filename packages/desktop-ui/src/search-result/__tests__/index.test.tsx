import React from 'react'
import SearchResult from '..'
import { render } from '../../../../../tests/utils'

describe('SearchResult', () => {
  it('should work', () => {
    const { container } = render(<SearchResult />)
    expect(container).toMatchSnapshot()
  })
})
