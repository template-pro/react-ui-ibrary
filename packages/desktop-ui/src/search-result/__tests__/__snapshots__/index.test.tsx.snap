// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult should work 1`] = `
<div>
  <div
    class="temp-pc-search-result"
  >
    <input
      class="ant-input ant-input-borderless"
      placeholder="请输入关键词查找"
      type="text"
      value=""
      width="100%"
    />
  </div>
</div>
`;
